src/feed.test.tsx

```diff
@@ -15,40 +64,112 @@ describe('Feed', () => {
   };
 
   test('renders a feed item for item', () => {
-    const items = [{
-      id: 'the-first-id',
-      title: 'The First Item',
-      description: 'This is the description of the first item.',
-    }, {
-      id: 'the-second-id',
-      title: 'The Second Item',
-      description: 'This is the description of the Second item.',
-    }] as FeedItemModel[];
-
-    const wrapper = subject({ items });
+    const wrapper = subject({ items: feedItemsTest.slice(0, 2) });
 
     expect(wrapper.find(FeedItem)).toHaveLength(2);
   });
 
   test('passes item props to FeedItem', () => {
-    const items = [{
-      id: 'the-first-id',
-      title: 'The First Item',
-      description: 'This is the description of the first item.',
-    }, {
-      id: 'the-second-id',
-      title: 'The Second Item',
-      description: 'This is the description of the Second item.',
-    }] as FeedItemModel[];
-
-    const wrapper = subject({ items });
+    const wrapper = subject({ items: feedItemsTest.slice(0, 2) });
 
     expect(wrapper.find(FeedItem).at(1).props()).toEqual(
       expect.objectContaining({
         id: 'the-second-id',
         title: 'The Second Item',
         description: 'This is the description of the Second item.',
-      }),
+      })
+    );
+  });
+
+  test('render the correct items initially', () => {
+    const wrapper = subject({ items: feedItemsTest });
+
+    expect(wrapper.find(FeedItem)).toHaveLength(Feed.pageSize);
+  });
+
+  test('rendered items, if items are added to the end of the array', () => {
+    const wrapper = subject({ items: feedItemsTest.slice(0, 7) });
+
+    expect(wrapper.find(FeedItem)).toHaveLength(Feed.pageSize);
+
+    wrapper.setProps({ items: feedItemsTest.slice(0, 9) });
+
+    expect(wrapper.find(FeedItem)).toHaveLength(Feed.pageSize);
+  });
+
+  test('render the correct data if one of the items has a property updated.', () => {
+    const newTitle = 'new title';
+    const wrapper = subject({ items: feedItemsTest });
+
+    expect(wrapper.find(FeedItem).at(0).prop('title')).toBe(
+      feedItemsTest[0].title
+    );
+
+    wrapper.setProps({
+      items: [
+        {
+          ...feedItemsTest[0],
+          title: newTitle,
+        },
+        ...feedItemsTest.slice(1, 9),
+      ],
+    });
+
+    expect(wrapper.find(FeedItem).at(0).prop('title')).toBe(newTitle);
+  });
+
+  test('render the correct items on fetch', () => {
+    const wrapper = subject({ items: feedItemsTest });
+
+    expect(wrapper.find(FeedItem)).toHaveLength(Feed.pageSize);
+
+    // fetch more data
+    (wrapper.instance() as Feed).fetchMoreData();
```

Comment:
this might seem small, but i think it's important: instead of calling this method on the feed component, let's use the seam created by the scroll component, as that is what is actually triggering the scroll:
```
wrapper.find(InfiniteScroll).simulate('next');
```
that ensures that we have the event wired up properly, that we are working to the interface of the component we're using, and also allows us to change the actual implementation, and still have the tests pass.

Reply:
The method use it  by the scroll component does't work with simulate (.simulate('next')). 
So I triggered it with the props call ( wrapper.find(InfiniteScroll).props().next() )  
Do you have any views ?